prusti-interface/src/environment/mir_utils/statement_at.rs

```diff
@@ -0,0 +1,17 @@
+use rustc_middle::mir;
+
+pub trait StatementAt<'tcx> {
+    fn statement_at(&self, location: mir::Location) -> Option<&mir::Statement<'tcx>>;
+}
+
+impl<'tcx> StatementAt<'tcx> for mir::Body<'tcx> {
+    fn statement_at(&self, location: mir::Location) -> Option<&mir::Statement<'tcx>> {
+        let block = &self[location.block];
+        if location.statement_index < block.statements.len() {
```

Comment:
```suggestion
        if location.statement_index != block.statements.len() {
```
It would be better to panic if the location for some reason is completely off.